modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/identity/integration/common/utils/UserStoreConfigUtils.java

```diff
@@ -40,6 +40,7 @@ public class UserStoreConfigUtils {
     public boolean waitForUserStoreDeployment(UserStoreConfigAdminServiceClient userStoreConfigAdminServiceClient,
                                               String domain) throws Exception {
 
+        Thread.sleep(5000);
         long waitTime = System.currentTimeMillis() + 30000; //wait for 45 seconds
```

Comment:
just asking to be aware, cannot we do this instead? 

```suggestion
        long waitTime = System.currentTimeMillis() + 35000; //wait for 35 seconds
```

Reply:
This wait time is the time we will be used to do the configuration check timeout. That means we are waiting maximum of 30 seconds. 

The first sleep is used to wait before the first configuration check API call. Since this is a user store adding it will never configure just after we send the create request. So if we remove the first 5s wait, it will lead to some additional API calls and intension here is to reduce those redundant API calls.